libs/wire-api/src/Wire/API/Routes/Public/Brig.hs

```diff
@@ -971,6 +971,7 @@ type ConnectionAPI =
     :<|> Named
            "create-connection"
            ( Summary "Create a connection to another user"
+               :> MakesFederatedCall 'Brig "get-users-by-ids"
```

Comment:
Should `CanThrow TeamsNotFederating` be added here?

Reply:
I don't think so. In Galley, we specify each individual error that a handler can throw and these are statically checked. In Brig, we have an ADT that a handler can throw, and these by the end of execution get converted to `StdError`, which is the code-label-message triple error response.

Reply:
I think it is useful for clients to know which status and labels to expect so that they can handle them.

Reply:
I agree, but with Brig we're not there (yet). Brig is in a half-finished state when it comes to migrating to Polysemy. With Polysemy we have the ability to list only specific errors an endpoint can return and have them checked by the compiler (as done in Galley). If I manually add this annotation to the endpoint, it will not be requested by the endpoint handler, and then we're back to manually annotating Swagger with errors an endpoint can return; such documentation is out of sync and out of date in no time.

Or am I getting this wrong?

Reply:
I think the ADT is specification enough, everything else has too much overhead.

Reply:
I'm not even aware; is that ADT somewhere added to Swagger?

Reply:
No. These ADTs are not reflected in Swagger. In this case, we have the `ConnectionError` error type, but before the endpoint returns, this error that is passed around within the `ConnectionM r` monad (just an `ExceptT`), is eventually converted to an `StdError`, e.g.:

```haskell
API.createConnection lself conn target !>> connError
```
where:
```haskell
connError :: ConnectionError -> Error
```